src/engine/shared/console.cpp

```diff
@@ -1129,3 +1093,45 @@ void CConsole::CResult::SetVictim(const char *pVictim)
 	else
 		m_Victim = std::clamp<int>(str_toint(pVictim), 0, MAX_CLIENTS - 1);
 }
+
+template<typename T>
+std::optional<T> CConsole::ColorParse(const char *pStr, float DarkestLighting)
```

Comment:
Why is it `template<typename T>` when it always only returns either `ColorHSLA` or `std::nullopt`
```suggestion
std::optional<ColorHSLA> CConsole::ColorParse(const char *pStr, float DarkestLighting)
```

Reply:
> `[It's the same as the other color parse function], instead of duplicating the function, you can just add another color type, if this wont be merged with the generic then casting the output isn't that henious`

Then I don't see this being relevant to ddnet, this color parsing function is used only in `GetColor`, which will now only have `return ColorParse`